new-site/src/components/Playground.js

```diff
@@ -0,0 +1,226 @@
+  const viewPortRef = useRef();
+  const copyButtonRef = useRef();
+  const [resetCount, setResetCount] = useState(0);
+  const textAreaId = `code-block-textarea-${language}-${resetCount}`;
+  const helperTextId = `code-block-helper-${language}-${resetCount}`;
+  const getTextArea = useCallback((el) => el.querySelector(`#${textAreaId}`), [textAreaId]);
+
+  const onFocusKeyDown = useCallback(
+    (event) => {
+      if (event.key === 'Enter' && viewPortRef.current && event.target === viewPortRef.current) {
+        event.preventDefault();
+        const textArea = getTextArea(viewPortRef.current);
+        textArea.focus();
+      }
+    },
+    [getTextArea],
+  );
+
+  const onTextAreaBlur = useCallback((e) => {
+    // Set viewport focus on textarea Esc keypress (no related target)
+    if (viewPortRef.current && !e.relatedTarget) {
+      viewPortRef.current.focus();
+    }
+  }, []);
+
+  const copy = () => {
+    if (viewPortRef.current && copyButtonRef.current) {
+      const textArea = getTextArea(viewPortRef.current);
+      textArea.focus();
+      textArea.select();
+
+      try {
+        document.execCommand('copy');
+        if (copyButtonRef.current) {
+          copyButtonRef.current.focus();
+        }
+      } catch (err) {
+        console.warn('Oops, unable to copy');
+      }
+      clearSelection();
+    }
+  };
+
+  const reset = () => {
+    onChange(initialCode);
+    setResetCount(resetCount + 1);
+    if (viewPortRef.current) {
+      viewPortRef.current.focus();
+    }
+  };
+
+  useEffect(() => {
+    if (viewPortRef.current) {
+      const textArea = getTextArea(viewPortRef.current);
+      textArea.setAttribute('aria-describedby', helperTextId);
+      textArea.setAttribute('tabIndex', '-1');
+      textArea.addEventListener('blur', onTextAreaBlur);
+    }
+  }, [getTextArea, helperTextId, onTextAreaBlur]);
+
+  return (
+    <>
+      <div className="playground-block-editor">
+        <div // eslint-disable-line jsx-a11y/no-static-element-interactions
+          className="playground-block-editor-viewport"
+          tabIndex={0} // eslint-disable-line jsx-a11y/no-noninteractive-tabindex
+          onKeyDown={onFocusKeyDown}
+          ref={viewPortRef}
+        >
+          <div className="playground-block-editor-code">
+            <div className="playground-block-editor-texts">
+              <label className="playground-block-editor-label" htmlFor={textAreaId}>
+                Editable code example
+              </label>
+              <span className="playground-block-editor-helper" id={helperTextId}>
+                Press Enter to start editing. Press Esc to stop editing.
+              </span>
+            </div>
+            <div className="playground-block-editor-scrollbox" tabIndex={-1}>
+              <LiveEditor
+                key={resetCount}
+                onChange={onChange}
+                style={{
+                  background: 'none',
```

Comment:
Should we put these styles to css file, or is it better to inline them here?

Reply:
The LiveEditor component seems to take only style property. It passes those styles to underlying textarea and pre components. I guess it is possible to give the className to parent LiveProvider component and then override the child element styles in css file 🤔

Reply:
Hmm, it might actually take to className as well. I'll try to use that one

Reply:
- [ ] replace live-editor inline styles with classname